modules/apps/frontend-js/frontend-js-walkthrough-web/test/js/Walkthrough.test.js

```diff
@@ -0,0 +1,251 @@
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import '@testing-library/jest-dom/extend-expect';
+import {cleanup, render, screen} from '@testing-library/react';
+import userEvents from '@testing-library/user-event';
+import {navigate} from 'frontend-js-web';
+import React from 'react';
+
+import Walkthrough from '../../src/main/resources/META-INF/resources/Walkthrough';
+import {
+	BOX_SHADOW_ELEMENT_MOCK,
+	DOM_STRUCTURE_FOR_PLACING_STEPS,
+	INVALID_NODE_SELECTOR_MOCK,
+	MULTI_PAGES_MOCK,
+	PAGE_MOCK,
+	PAGE_WITH_PREVIOUS_MOCK,
+} from './__fixtures__/walkthroughMock';
+
+function setupDocument() {
+	const domStructureReference = document.createElement('div');
+
+	domStructureReference.innerHTML = DOM_STRUCTURE_FOR_PLACING_STEPS;
+
+	domStructureReference.id = 'test-structure';
+
+	document.body.appendChild(domStructureReference);
+
+	return () => {
+		window.localStorage.clear();
+
+		const nodeToBeRemoved = document.getElementById('test-structure');
+
+		document.body.removeChild(nodeToBeRemoved);
+
+		nodeToBeRemoved.remove();
+	};
+}
+
+function renderWalkthrough(props) {
+	return render(<Walkthrough {...props} />, {
+		baseElement: document.getElementById('app_root'),
+	});
+}
+
+jest.mock('frontend-js-web', () => ({
+	navigate: jest.fn((url) => url),
+}));
+
+const USER_ID = 42;
+
+describe('Walkthrough', () => {
+	window.themeDisplay = {
+		getLayoutRelativeURL: jest.fn(() => '/home'),
+		getUserId: jest.fn(() => USER_ID),
+	};
+
+	afterEach(() => {
+		cleanup();
+	});
+
+	beforeEach(() => {
+		jest.restoreAllMocks();
+	});
+
+	it('renders', () => {
+		const cleanUp = setupDocument();
+
+		const {container} = renderWalkthrough(PAGE_MOCK);
+
+		expect(container).toBeInTheDocument();
+
+		cleanUp();
+	});
+
+	it(`when clicking on Next, it navigates to the given URL of 'next'`, () => {
+		const cleanUp = setupDocument();
+
+		const {getByLabelText} = renderWalkthrough(MULTI_PAGES_MOCK);
+
+		const hotspot = getByLabelText('start-the-walkthrough');
+
+		userEvents.click(hotspot);
+
+		userEvents.click(screen.getByText('ok'));
+
+		expect(navigate).toBeCalledWith(MULTI_PAGES_MOCK.steps[0].next);
+
+		cleanUp();
+	});
+
+	it(`when clicking on Previous, it navigates to the given URL of 'previous'`, () => {
+		const cleanUp = setupDocument();
+
+		const {getByLabelText} = renderWalkthrough(PAGE_WITH_PREVIOUS_MOCK);
+
+		const hotspot = getByLabelText('start-the-walkthrough');
+
+		userEvents.click(hotspot);
+
+		userEvents.click(screen.getByText('ok'));
+
+		userEvents.click(screen.getByText('previous'));
```

Comment:
Maybe another option here would also be to define the current step in localStorage in the last step and opened, so we can define the step with next and previous correctly.